tests/unit.helpers.tsx

```diff
@@ -5,9 +5,35 @@ import os from 'os'
 import jwt from 'jsonwebtoken'
 import { headers } from 'next/headers.js'
 import { readTestSupportFile } from './common.helpers'
+import { render } from '@testing-library/react'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { MantineProvider } from '@mantine/core'
+import { ModalsProvider } from '@mantine/modals'
+import { theme } from '@/theme'
+import { ReactElement } from 'react'
+
+const createTestQueryClient = () =>
+    new QueryClient({
+        defaultOptions: {
+            queries: {
+                retry: false,
+            },
+        },
+    })
+
+export function renderWithProviders(ui: ReactElement) {
+    const testQueryClient = createTestQueryClient()
+
+    return render(
+        <QueryClientProvider client={testQueryClient}>
+            <MantineProvider theme={theme}>
+                <ModalsProvider>{ui}</ModalsProvider>
+            </MantineProvider>
+        </QueryClientProvider>,
+    )
+}
```

Comment:
instead of this, we can use the 2nd context argument of `render` along with the `TestingProvidersWrapper`   Example: https://github.com/safeinsights/management-app/blob/main/src/app/researcher/study/request/%5BmemberIdentifier%5D/form.test.tsx#L16-L17

I do see this being a handy abstraction though so maybe we still export it.  up to you I guess

Reply:
yeah i suppose this is totally a stylistic thing right?

These would be functionally equivalent 🤔 
```
        const {container} = render(
            <ManageMemberPage params={props.params} />,
            TestingProvidersWrapper
        )

        const { container } = renderWithProviders(await ManageMemberPage(props))
```

Reply:
im happy to go either way on this one! if we want to leave it in and see how it goes organically that works for me too